File: Python/2309. Greatest English Letter in Upper and Lower Case.py
class Solution:
    def greatestLetter(self, s: str) -> str:
#         s, m = set(s), ' '

#         for i in s:
#             if chr(ord(i) + 32) in s and ord(i) > ord(m): m = i
        
#         return m if m != ' ' else ''

        for i in range(90, 64, -1):
            if chr(i) in s and chr(i + 32) in s: return chr(i)
        
        return ''

File: Java/2309. Greatest English Letter in Upper and Lower Case.java
class Solution {
    public String greatestLetter(String s) {
//         char m = ' ';

//         for(char i: s.toCharArray()){
//             if(s.indexOf(i + 32) != -1 && i > m) m = i;
//         }

//         return String.valueOf(m == ' ' ? "" : m);

        for(char i = 'Z'; i >= 'A'; i--){
            if(s.indexOf(i) != -1 && s.indexOf(i + 32) != -1) return i + "";
        }

        return "";
    }
}
